src/puya/ir/optimize/arithmetic.py

```diff
@@ -241,7 +241,13 @@ def try_simplify_arithmetic_ops(value: models.ValueProvider) -> models.ValueProv
                     case AVMOp.mul:
                         c = a_const * b_const
                     case AVMOp.div_floor:
+                        if b_const == 0:
```

Comment:
We had some checks like this previously, but I think they were removed because it is possible these operations are guarded by a conditional that would prevent their evaluation at runtime

Reply:
ah of course, I'll drop the `raise`